### PR TITLE
Update Discord contact point documentation for use_discord_username type

### LINKED ISSUE #94893: Documentation feedback: /docs/sources/alerting/set-up/provision-alerting-resources/file-provisioning/index.md
[Documentation for alerting provisioning](https://grafana.com/docs/grafana/latest/alerting/set-up/provision-alerting-resources/file-provisioning/#discord) says that **use_discord_username** setting for discord contact point has type of string.
```
type: discord
settings:
  ...
  # <string>
  use_discord_username: Grafana
  ...
```
In fact **use_discord_username** setting has boolean type
Providing string results configuration error:
```
 integration \"mycontactpoint\" (UID myuid) of type \"discord\": failed to unmarshal settings: json: cannot unmarshal string into Go struct field Config.use_discord_username of type bool"
Error: ✗ failure to map file alert-resources.yml: failure parsing contact points: mycontactpoint: failed to validate integration "mycontactpoint" (UID myuid) of type "discord": failed to unmarshal settings: json: cannot unmarshal string into Go struct field Config.use_discord_username of type bool
```

I think that documentation should be
```
type: discord
settings:
  ...
  # <bool>
  use_discord_username: false
  ...
```

### PR DESCRIPTION
**Pull request by chitvs**
**What is this feature?**

This pull request updates the documentation for the Discord contact point in the alerting setup. It corrects the data type of the `use_discord_username` setting from `<string>` to `<bool>`.

**Why do we need this feature?**

The current documentation specifies `use_discord_username` as a `<string>`, but it should be a `<bool>`. This update helps prevent configuration errors for users setting up Discord contact points in alerting.

**Who is this feature for?**

This update is for Grafana users who are configuring alerting contact points with Discord. It ensures accurate documentation, which is crucial for correctly setting up alerting integrations.

**Which issue(s) does this PR fix?**

Fixes #94893

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Comment by chitvs** [comment]
Hi @brendamuir, is there anything else needed to finalize this PR? Thanks!

Best,
chitvs

**Review by brendamuir — APPROVED** [review]

**Comment by chitvs** [comment]
Glad to see you approved changes, now We are waiting for checks approval or something else?

**Pull request merged**
